tests/pybind11_tests.h

```diff
@@ -96,3 +96,23 @@ void ignoreOldStyleInitWarnings(F &&body) {
     )",
              py::dict(py::arg("body") = py::cpp_function(body)));
 }
+
+class TestContext {
```

Comment:
Could you please add a comment:

// See PR #5419 for background.

(That'll make it much faster for people to find out what this code is about.)